modules/ml/src/main/java/org/apache/ignite/ml/tree/DecisionTreeConditionalNode.java

```diff
@@ -45,17 +48,29 @@
      * @param threshold Threshold.
      * @param thenNode Node that will be used in case tested value is greater then threshold.
      * @param elseNode Node that will be used in case tested value is not greater then threshold.
+     * @param missingNode Node that will be used in case tested value is not presented.
      */
-    DecisionTreeConditionalNode(int col, double threshold, DecisionTreeNode thenNode, DecisionTreeNode elseNode) {
+    public DecisionTreeConditionalNode(int col, double threshold, DecisionTreeNode thenNode, DecisionTreeNode elseNode,
+        DecisionTreeNode missingNode) {
         this.col = col;
         this.threshold = threshold;
         this.thenNode = thenNode;
         this.elseNode = elseNode;
+        this.missingNode = missingNode;
     }
 
     /** {@inheritDoc} */
     @Override public Double apply(Vector features) {
-        return features.get(col) > threshold ? thenNode.apply(features) : elseNode.apply(features);
+        double val = features.get(col);
+
+        if (Double.isNaN(val)) {
```

Comment:
maybe we save old logic of NaN processing?
like: 

if (Double.isNaN(val) && missingNode != null) 
   return missingNode.apply(features);
else 
   return val > threshold ? thenNode.apply(features) : elseNode.apply(features);

Reply:
in terms of compatibility with old version?

Reply:
The result of old logic was a lot of hidden mistakes (NaN values). Current solution allows to eliminate them. Do you think that it's better to hide such mistakes?